### ISSUE TITLE
Include() results in two separate query to fetch data

### ISSUE THREAD

**Issue by bidelakbari** [dotnet/efcore#7415]
### Steps to reproduce
I have Blog and Post entities. when I enable eager loading on Blog I see two separate queries in SQL Profiler.

_So, for this:_
```c#
foreach (var blg in db.Blogs.Include(x => x.Posts))
{
    Console.WriteLine(blg.Posts.Count);
}
```

_I get this:_
```TSQL
SELECT [x].[BlogId], [x].[CreatedOn], [x].[Name]
FROM [Blogs] AS [x]
ORDER BY [x].[BlogId]
```

_and this in Sql Profiler:_
```TSQL
SELECT [p].[PostId], [p].[BlogId], [p].[Body], [p].[CreatedOn], [p].[Title]
FROM [Posts] AS [p]
WHERE EXISTS (
    SELECT 1
    FROM [Blogs] AS [x]
    WHERE [p].[BlogId] = [x].[BlogId])
ORDER BY [p].[BlogId]
```

I believe the first one is redundant.

### Further technical details
EF Core version: 1.1.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer 1.1.0
Operating system: Win 10
IDE: Visual Studio 2015

**Comment by bidelakbari** [comment]
Discard this issue please. It had to do with an incomplete db migration.

**Comment by gdoron** [comment]
You should probably read the discussion in this thread:
#1833 
(I spammed it with an off-topic discussion 😿, but it has the answer to your question )

TL;DR; It's by design.

**Comment by rowanmiller** [comment]
As @gdoron noted, this is by design
